lib/kvs/kvs_cursor.c

```diff
@@ -1146,11 +1146,18 @@ kvs_cursor_val_copy(
     const void            **val_out,
     size_t                 *vlen_out)
 {
-    struct kvs_cursor_impl *cur = cursor_h2r(cursor);
+    struct kvs_cursor_impl *cur;
+    struct kvs_vtuple *     vt;
+    uint                    clen;
+    merr_t                  err = 0;
+
+    if (!cursor)
+        return merr(EINVAL);
```

Comment:
Can that be an invariant?  Why would cursor ever be nil?

Reply:
I thought we agreed that at API boundaries we wouldn't use invariants.

Reply:
Sigh, this review tool sucks, don't even know what function this is...
I would make the exception if this is an internal interface that is called frequently...

Reply:
Greg, you should grab VSCode and use it as your review tool for PRs. Allows you to make comments and approve all from the editor. Can still use emacs or whatever for editing :).

Reply:
This is called in the cursor read path.